src/shared_utilities/drivers.jl

```diff
@@ -586,3 +587,70 @@ function initialize_drivers(
     radiation_drivers = initialize_drivers(r, coords)
     merge(atmos_drivers, radiation_drivers)
 end
+
+"""
+    make_update_drivers(a::Union{AbstractAtmosphericDrivers, Nothing},
+                          r::Union{AbstractRadiativeDrivers, Nothing},
+                         )
+
+Creates and returns a function which updates the atmospheric
+and radiative forcing variables ("drivers").
+"""
+function make_update_drivers(
+    a::Union{AbstractAtmosphericDrivers, Nothing},
+    r::Union{AbstractRadiativeDrivers, Nothing},
+)
+    update_atmos! = make_update_drivers(a)
+    update_radiation! = make_update_drivers(r)
+    function update_drivers!(drivers, t)
+        update_atmos!(drivers, t)
+        update_radiation!(drivers, t)
+    end
+    return update_drivers!
+end
+
+"""
+    make_update_drivers(d::Nothing)
+
+Creates and returns a function which updates the driver variables
+in the case of no driver variables.
+"""
+make_update_drivers(d::Nothing) = (p, t) -> nothing
+
+"""
+    make_update_drivers(a::PrescribedAtmosphere{FT}) where {FT}
+
+Creates and returns a function which updates the driver variables
+in the case of a PrescribedAtmosphere at a point.
+"""
+function make_update_drivers(a::PrescribedAtmosphere{FT}) where {FT}
+    function update_drivers!(drivers, t)
+        drivers.P_liq .= FT(a.liquid_precip(t))
+        drivers.P_snow .= FT(a.snow_precip(t))
+        drivers.T .= FT(a.T(t))
+        drivers.P .= FT(a.P(t))
+        drivers.u .= FT(a.u(t))
+        drivers.q .= FT(a.q(t))
+        drivers.c_co2 .= FT(a.c_co2(t))
+    end
+    return update_drivers!
+end
+
+"""
+    make_update_drivers(r::PrescribedRadiativeFluxes{FT}) where {FT}
+
+Creates and returns a function which updates the driver variables
+in the case of a PrescribedRadiativeFluxes at a point.
+"""
+function make_update_drivers(r::PrescribedRadiativeFluxes{FT}) where {FT}
+    function update_drivers!(drivers, t)
+        drivers.SW_d .= FT(r.SW_d(t))
+        drivers.LW_d .= FT(r.LW_d(t))
+        if ~(r.θs isa Nothing)
```

Comment:
I think it's recommended to use `isnothing(var)` to check if something is nothing, but this is fine too

Reply:
I changed this!